### PR TITLE
[FEATURE] Display Project Info from guides.xml in Footer

### PR DESCRIPTION
**Pull request by linawolf**
Resolves https://github.com/TYPO3-Documentation/render-guides/issues/316
Depends on https://github.com/TYPO3-Documentation/render-guides/pull/326

### CODE REVIEW

File: Documentation/guides.xml [hunk 1]
```diff
@@ -14,6 +14,10 @@
         edit-on-github-branch="main"
         interlink-shortcode="t3docsRenderGuides"
         typo3-core-preferred="stable"
+        project-home="https://github.com/TYPO3-Documentation/render-guides"
```
Comment (brotkrueml) [inline Documentation/guides.xml:17]: Until now, the attributes are sorted alphabetically. Can we stay that way?

File: packages/typo3-docs-theme/resources/template/structure/layoutParts/footer.html.twig [hunk 2]
```diff
@@ -3,11 +3,22 @@
     <div class="frame frame-ruler-before frame-background-dark">
         <div class="frame-container">
             <div class="frame-inner">
-                <ul class="footer-simplemenu">
-                    <li><a href="https://typo3.org/community/teams/documentation/#c9886" title="Contact"><span>Contact</span></a></li>
-                    <li><a href="https://github.com/TYPO3-Documentation/DocsTypo3Org-Homepage/issues" rel="nofollow noopener" title="Issues"><span>Issues</span></a></li>
-                    <li><a href="https://github.com/TYPO3-Documentation/DocsTypo3Org-Homepage" rel="nofollow noopener" title="Repository"><span>Repository</span></a></li>
-                </ul>
+                {% if env.projectNode.variables.project_home or env.projectNode.variables.project_contact or env.projectNode.variables.project_contact or env.projectNode.variables.project_repository -%}
```
Comment (brotkrueml) [inline packages/typo3-docs-theme/resources/template/structure/layoutParts/footer.html.twig:6]:
```suggestion
                {% if env.projectNode.variables.project_home or env.projectNode.variables.project_contact or env.projectNode.variables.project_issues or env.projectNode.variables.project_repository -%}
```

File: packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public function __construct(
+        private readonly Typo3DocsThemeSettings $themeSettings,
+    ) {}
+
+    public function __invoke(PostProjectNodeCreated $event): void
+    {
+        $projectNode = $event->getProjectNode();
+        foreach ($this->themeSettings->getAllSettings() as $key => $setting) {
+            if ($setting !== '') {
```
Comment (brotkrueml) [inline packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php:21]: Maybe a trim around `$setting`. It is an edge case.

File: packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                $projectNode->addVariable($key, new PlainTextInlineNode(strval($setting)));
```
Comment (brotkrueml) [inline packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php:22]: You check for an non-empty string above, but then cast it to a string with strval(). Either the check should be adjusted (maybe null?) or the strval is superfluous.

File: packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php [hunk 5]
```diff
@@ -23,4 +23,12 @@ public function getSettings(string $key, string $default = ''): string
         }
         return $this->settings[$key];
     }
+
+    /**
+     * @return array<string, string>
+     */
+    public function getAllSettings(): array
```
Comment (brotkrueml) [inline packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php:30]: Can we have unit tests for this class (as you introduce a new method)?

**Review by brotkrueml — COMMENTED** [review]
Is the patch about "contact info" or "project infos"?

File: packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php (hunk 5 above)
Comment (linawolf) [reply packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php:30]: You want unit tests for the getters and setters on a data object? I find that quite superfluous

**Comment by linawolf** [comment]
> Is the patch about "contact info" or "project infos"?

I dont really see the difference, which one do you suggest?

File: packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php (hunk 4 above)
Comment (linawolf) [reply packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php:22]: true, even though the $settings where supposed to be `array<string, string>` still a boolean value was passed. (wondering why phpstan didnt catch this). Anyway, I fix it now

**Comment by brotkrueml** [comment]
> > Is the patch about "contact info" or "project infos"?
> 
> I dont really see the difference, which one do you suggest?

Well, the project tag has 

> > Is the patch about "contact info" or "project infos"?
> 
> I dont really see the difference, which one do you suggest?

Well, we have contact, home, issues, repository, so I would prefer project as the umbrella.

File: packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php (hunk 5 above)
Comment (brotkrueml) [reply packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php:30]: You have logic in it (isset check). But I am fine to move that to a dedicated PR.

File: Documentation/guides.xml (hunk 1 above)
Comment (linawolf) [reply Documentation/guides.xml]: Ack

File: packages/typo3-docs-theme/resources/template/structure/layoutParts/footer.html.twig (hunk 2 above)
Comment (linawolf) [reply packages/typo3-docs-theme/resources/template/structure/layoutParts/footer.html.twig]: Good catch!

File: packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php (hunk 5 above)
Comment (linawolf) [reply packages/typo3-docs-theme/src/Settings/Typo3DocsThemeSettings.php:30]: Feel free to provide a PR

**Comment by linawolf** [comment]
@brotkrueml can we merge this?

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**